### PR TITLE
fixed matching algirithm

### PR DESCRIPTION
**Pull request by EraKin575**
## Related Issue
  - Info about Issue or bug

Closes: #908 

#### Describe the changes you've made
fixed the issue for matching algorithm

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by EraKin575** [comment]
recheck

**Comment by EraKin575** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by charankamarapu** [comment]
Hi @EraKin575 ping here once you have done with the changes I will review it.

File: pkg/proxy/integrations/genericParser/util.go [hunk 1]
```diff
@@ -24,58 +24,46 @@ func PostgresDecoder(encoded string) ([]byte, error) {
...
 					encoded, _ = PostgresDecoder(mock.Spec.GenericRequests[requestIndex].Message[0].Data)
 				}
 
-				if string(encoded) == string(reqBuff) || mock.Spec.GenericRequests[requestIndex].Message[0].Data == bufStr {
-					log.Debug("matched in first loop")
-					tcsMocks = append(tcsMocks[:idx], tcsMocks[idx+1:]...)
-					h.SetTcsMocks(tcsMocks)
-					return true, mock.Spec.GenericResponses
+				// Compare the encoded data
+				if string(encoded) != string(reqBuff) && mock.Spec.GenericRequests[requestIndex].Message[0].Data != bufStr {
```
Comment (charankamarapu) [inline pkg/proxy/integrations/genericParser/util.go:43]: shouldn't this be "or" condition, even if any one condition is false that means both requests are not same right ?

**Review by charankamarapu — DISMISSED** [review]
Ship it!

**Review by charankamarapu — COMMENTED** [review]
Please resolve the comments

**Pull request closed without merging**